src/librbd/ImageWatcher.cc

```diff
@@ -844,7 +844,6 @@ void ImageWatcher::handle_payload(const FlattenPayload &payload,
 
 void ImageWatcher::handle_payload(const ResizePayload &payload,
 				  bufferlist *out) {
-  RWLock::RLocker l(m_image_ctx.owner_lock);
```

Comment:
@dillaman any more comments on this? My concern here is that if we take the owner lock at the beginning of this function,  when we calls the async resize state machine (async_resize), we'll again take the owner lock twice.

Or could we take the owner lock at the beginning, and release it after inserting the request into the async pending list (which is before calling async resize state machine)?